proof/courier.go

```diff
@@ -90,20 +55,123 @@ type CourierHarness interface {
 // receiver can use this to fetch a proof from the sender.
 //
 // TODO(roasbeef): FileSystemCourier, RpcCourier
-type Courier[Addr any] interface {
+type Courier interface {
 	// DeliverProof attempts to delivery a proof to the receiver, using the
 	// information in the Addr type.
-	DeliverProof(context.Context, Addr, *AnnotatedProof) error
+	DeliverProof(context.Context, Recipient, *AnnotatedProof) error
```

Comment:
I don't think we want to hard code the `Recipient` struct here. Instead we can lift that information up into the `Courier` implementation itself. In other words, when the instance is created, it already knows "who" it's sending to, and then later you bind the "what". 

For hash mail "who" is the hash mail addr, and the what is the proof. 

For multi-verses, the "who" is the server+outpoint info, and the what is the proof itself. 

Thinking about it a bit more, for the multi-verse instance, we already have the outpoint, as it's contained in the last state transition in the prof.

Reply:
I understand, will move recipient into the new handle creator.